sim/core/spell.go

```diff
@@ -178,6 +177,22 @@ func (unit *Unit) RegisterSpell(config SpellConfig) *Spell {
 		config.Cast.DefaultCast.AfterCastDelay = 0
 	}
 
+	if (config.DamageMultiplier != 0 || config.ThreatMultiplier != 0) && config.ProcMask == ProcMaskUnknown {
+		panic("ProcMask for spell " + config.ActionID.String() + " not set")
+	}
+
+	if (config.DamageMultiplier != 0 || config.ThreatMultiplier != 0) && config.SpellSchool == SpellSchoolNone {
+		panic("SpellSchool for spell " + config.ActionID.String() + " not set")
```

Comment:
I feel like there are spells somewhere with no school. I know gift of the naaru (draenei racial) has no school, for example, although we haven't implemented it yet.

Reply:
This check is pre-existing, so I'd leave it "as is". Adding more static checks here helps moving stuff out of the critical path, so this is a good thing.